services/src/main/java/org/keycloak/services/resources/LoginActionsService.java

```diff
@@ -815,6 +815,7 @@ public Response authenticateOnly() throws AuthenticationFlowException {
             protected Response authenticationComplete() {
                 if (firstBrokerLogin) {
                     authSession.setAuthNote(AbstractIdpAuthenticator.FIRST_BROKER_LOGIN_SUCCESS, identityProviderAlias);
+                    event.success();
```

Comment:
Why not send the event "IDENTITY_PROVIDER_POST_LOGIN" at the same time ?

Reply:
I think only one of the two should be triggered, not both, as you can see here:  https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/services/resources/LoginActionsService.java#L755

however, thanks for drawing my attention to that. Now I need to slightly change the `event.success()` call to be outside of the if.